### ISSUE TITLE
Listing memorized rolls

### ISSUE THREAD

**Issue by Lithimlin** [ArtemGr/Sidekick#136]
There is currently only the first line being output which reads:
`@User, you have the following rolls memorized:`

**Comment by Lithimlin** [comment]
After some testing it seems like this is a bug that only occurs in like 50% of the cases.
The embedded Link seems to not always work well

**Comment by Lithimlin** [comment]
The problem was that the Bot didn't have permissions to embed links.
Might wanna make people aware of that.

**Comment by ArtemGr** [comment]
> There is currently only the first line being output which reads:
> @User, you have the following rolls memorized:

This sounds a bit like #132, could they be connected?

> The embedded Link seems to not always work well

What link do you mean?
I wasn't aware we're using links in the memorized rolls.

**Comment by Lithimlin** [comment]
It appears that the list of saved rolls is in form of an embedding

**Comment by ArtemGr** [comment]
Yes, it is, but it shouldn't be using links. I'll give it a second look.

**Comment by ArtemGr** [comment]
Not using any links here, 
```
        for chunk in rollz.chunks (25) {  // Discord seems to limit embeds to 25 fields.
          try_s! (bot.send_embed (message.channel_id(), &with_mention, &|mut eb| {
            eb = eb.fields (|mut efb| {
              for &(ref name, ref expression) in chunk {
                efb = efb.field (&name, &expression, false)}
              efb});
            eb}))}}
```

Maybe permission is a misnomer.

@Lithimlin , so you're saying that for embeds to work we need the EMBED_LINKS ("Links sent by users with this permission will be auto-embedded") [permission](https://discordapp.com/developers/docs/topics/permissions)?

**Comment by Lithimlin** [comment]
Yes, it seems to be that way. Without this permission, no further messages will be shown (such as how many roll saves you have left and the aforementioned list of saved rolls).

**Comment by dalek955** [comment]
I have this problem on my server too, and the "Bots" role there does have the Embed Links permission.

**Comment by dalek955** [comment]
That link doesn't work. How do I PM you and what do you want to know?

**Comment by ArtemGr** [comment]
The link does work for me, but you can give me the user ID that fails to see the rolls instead.
Preferably the digital User ID that's available with "Copy ID".

![](https://i.imgur.com/1NkAU0Q.png)

**Comment by dalek955** [comment]
My user ID is 355565895080345602When I ask Sidekick to list memorized rolls, this happens:



-----Original Message-----

**Comment by vagueGM** [comment]
I have just tested this and the bot does need the 'Embed Links' permission.
The user also need to have the 'Show website preview info from links pasted into chat' setting turned on under 'Text & Images' in 'User Settings'.
Hope this helps.

**Comment by ArtemGr** [comment]
Can try printing the labels differently if that helps

**Comment by vagueGM** [comment]
As suggested above, maybe just mentioning it in the documentation is enough?
